pkg/file/builder.go

```diff
@@ -182,6 +183,44 @@ func (b *stateBuilder) addConsumerGroupPlugins(
 	return nil
 }
 
+func (b *stateBuilder) partials() {
+	if b.err != nil {
+		return
+	}
+
+	for _, p := range b.targetContent.Partials {
+		var (
+			partial *state.Partial
+			err     error
+		)
+		if !utils.Empty(p.Name) {
+			partial, err = b.currentState.Partials.Get(*p.Name)
+		}
+		if utils.Empty(p.ID) {
+			if errors.Is(err, state.ErrNotFound) || partial == nil {
+				p.ID = uuid()
+			} else if err != nil {
+				b.err = err
+				return
+			} else {
+				p.ID = kong.String(*partial.ID)
```

Comment:
Can `partial.ID` be `nil` here? That would cause a panic.

As far as I can see this could happen when `utils.Empty(p.Name)` returns `true` right?

Reply:
I have added a check on L200 to see if partial is nil. If it is, we generate an id.
Added another one to check for the ID as well. It is very less likely to happen but better to catch it than panic.